adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/DefaultDeviceResolver.java

```diff
@@ -240,4 +256,24 @@ private PreSharedKeyDeviceIdentity getHandshakeIdentity(final Span span, final S
         final String splitRegex = config.isSingleTenant() ? null : config.getIdSplitRegex();
         return PreSharedKeyDeviceIdentity.create(span, identity, splitRegex);
     }
+
+    @Override
+    public boolean hasEcdhePskSupported() {
+        return true;
+    }
+
+    @Override
+    public PskSecretResult requestPskSecretResult(final ConnectionId cid, final ServerNames serverName,
+            final PskPublicInformation identity, final String hmacAlgorithm, final SecretKey otherSecret,
+            final byte[] seed) {
+
+        context.runOnContext((v) -> loadCredentialsForDevice(cid, identity));
+        return null;
+    }
+
+    @Override
+    public void setResultHandler(final PskSecretResultHandler resultHandler) {
```

Comment:
which component invokes this method and when?

Reply:
The DTLSConnector in initialization.

Reply:
So this is only set once and not for each invocation of *requestPskSecretResult*, right?

Reply:
yes.

Reply:
Just to mention: 
otherwise I would have included that callback in the call.

Reply:
That is the answer I had hoped for/expected ;-)